src/wallet/asyncrpcoperation_sendmany.cpp

```diff
@@ -191,791 +149,514 @@ void AsyncRPCOperation_sendmany::main() {
-            txValues.selectedUTXOCoinbase = true;
-        }
-        selectedUTXOAmount += out.Value();
-        selectedTInputs.emplace_back(out);
-        if (selectedUTXOAmount >= txValues.targetAmount) {
-            // Select another utxo if there is change less than the dust threshold.
-            dustChange = selectedUTXOAmount - txValues.targetAmount;
-            if (dustChange == 0 || dustChange >= dustThreshold) {
-                break;
-            }
+            return true;
         }
     }
 
-    t_inputs_ = selectedTInputs;
-    txValues.t_inputs_total = selectedUTXOAmount;
-
-    if (txValues.t_inputs_total < txValues.targetAmount) {
-        throw JSONRPCError(RPC_WALLET_INSUFFICIENT_FUNDS,
-                           strprintf("Insufficient transparent funds, have %s, need %s",
-                                     FormatMoney(txValues.t_inputs_total), FormatMoney(txValues.targetAmount)));
-    }
-
-    // If there is transparent change, is it valid or is it dust?
-    if (dustChange < dustThreshold && dustChange != 0) {
-        throw JSONRPCError(RPC_WALLET_INSUFFICIENT_FUNDS,
-                           strprintf("Insufficient transparent funds, have %s, need %s more to avoid creating invalid change output %s (dust threshold is %s)",
-                                     FormatMoney(txValues.t_inputs_total), FormatMoney(dustThreshold - dustChange), FormatMoney(dustChange), FormatMoney(dustThreshold)));
-    }
-
-    // update the transaction with these inputs
-    if (isUsingBuilder_) {
-        for (const auto& out : t_inputs_) {
-            const CTxOut& txOut = out.tx->vout[out.i];
-            builder_.AddTransparentInput(COutPoint(out.tx->GetHash(), out.i), txOut.scriptPubKey, txOut.nValue);
-        }
-    } else {
-        CMutableTransaction rawTx(tx_);
-        for (const auto& out : t_inputs_) {
-            rawTx.vin.push_back(CTxIn(COutPoint(out.tx->GetHash(), out.i)));
-        }
-        tx_ = CTransaction(rawTx);
-    }
-    return true;
+    return false;
 }
 
-bool AsyncRPCOperation_sendmany::find_unspent_notes() {
-    std::vector<SproutNoteEntry> sproutEntries;
-    std::vector<SaplingNoteEntry> saplingEntries;
-    // TODO: move this to the caller
-    auto zaddr = KeyIO(Params()).DecodePaymentAddress(fromaddress_);
-    std::optional<AddrSet> noteFilter = std::nullopt;
-    if (zaddr.has_value()) {
-        noteFilter = AddrSet::ForPaymentAddresses(std::vector({zaddr.value()}));
-    }
-    pwalletMain->GetFilteredNotes(sproutEntries, saplingEntries, noteFilter, mindepth_);
-
-    // If using the TransactionBuilder, we only want Sapling notes.
-    // If not using it, we only want Sprout notes.
-    // TODO: Refactor `GetFilteredNotes()` so we only fetch what we need.
-    if (isUsingBuilder_) {
-        sproutEntries.clear();
-    } else {
-        saplingEntries.clear();
+void SpendableInputs::LogInputs(const AsyncRPCOperationId& id) const {
+    for (const auto& utxo : utxos) {
```

Comment:
Check that this can't be a performance bottleneck. Some nodes have *lots* of UTXOs. Now that I think about it, this may be why we didn't originally do this logging.